cmd/grafana-app-sdk/project.go

```diff
@@ -609,34 +611,109 @@ func projectAddPluginAPI[G anyGenerator](generator G, repo, generatedAPIModelsPa
 	return nil
 }
 
-//
 // Frontend plugin
 //
-
-func addComponentFrontend(projectRootPath string, pluginID string, promptForOverwrite bool) error {
+//nolint:revive
+func addComponentFrontend(projectRootPath string, pluginID string) error {
 	// Check plugin ID
 	if pluginID == "" {
 		return fmt.Errorf("plugin-id is required")
 	}
 
-	err := writeStaticFrontendFiles(filepath.Join(projectRootPath, "plugin"), promptForOverwrite)
+	if !isCommandInstalled("yarn") {
+		return fmt.Errorf("yarn must be installed to add the frontend component")
+	}
+
+	args := []string{"create", "@grafana/plugin", "--pluginType=app", "--hasBackend=true", "--pluginName=tmp", "--orgName=tmp"}
```

Comment:
At this point, I was just trying to mimic how the current frontend codegen acts (which always adds a backend), with the intention of cleaning this up in a future PR (because this is sort of a bug).